liveblog.php

```diff
@@ -451,14 +461,14 @@ public static function is_liveblog_post( $post_id = null ) {
 	 * @return bool
 	 */
 	public static function is_viewing_liveblog_post() {
-		return (bool) ( is_single() && self::is_liveblog_post() );
+		return (bool) ( is_singular() && self::is_liveblog_post() );
 	}
 
 	/**
 	 * One of: 'enable', 'archive', false.
 	 */
 	public static function get_liveblog_state( $post_id = null ) {
-		if ( ! is_single() && ! is_admin() && ! self::$is_rest_api_call) {
+		if ( !is_singular() && ! is_admin() && ! self::$is_rest_api_call) {
```

Comment:
Same as above - needs more precision.

Reply:
I think we could use get_post_type() here.
https://developer.wordpress.org/reference/functions/get_post_type/

Something like:
...
if ( !is_singular( get_post_type( $post_id ) ) ... )

Reply:
`is_singular` will accept post types as the parameter. You can pass the static `$supported_post_types` variable to the function instead.